include/wil/windowing.h

```diff
@@ -0,0 +1,78 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_WINDOWING_INCLUDED
+#define __WIL_WINDOWING_INCLUDED
+
+#include <WinUser.h>
+
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+namespace wil
+{
+    namespace details
+    {
+        template<typename TCallback>
+        BOOL CALLBACK LambdaEnumWindowsProc(HWND data, LPARAM lParam) noexcept
+        {
+            const TCallback* pCallback = reinterpret_cast<const TCallback*>(lParam);
+            using result_t = decltype((*pCallback)(data));
+#ifdef __cpp_if_constexpr

```

Comment:
```suggestion
#ifdef __cpp_if_constexpr
            using result_t = decltype((*pCallback)(data));
```
I believe that should resolve your build failure